tools/futhark/Dockerfile

```diff
@@ -1,22 +1,26 @@
-FROM python:3.11-slim
+FROM ghcr.io/astral-sh/uv:python3.11-bookworm-slim
+WORKDIR /gradbench
 
-RUN apt update
-RUN apt install -y xz-utils build-essential git
+# Install build dependencies.
+RUN apt-get update && apt-get install -y build-essential git xz-utils
 
-RUN pip install futhark-data futhark-server dataclasses-json
+# `COPY` the minimal set of files for `uv sync`.
+COPY pyproject.toml uv.lock ./
+COPY python/gradbench/pyproject.toml python/gradbench/
+COPY python/gradbench/gradbench/__init__.py python/gradbench/gradbench/
+RUN uv sync --no-default-groups --group dataclasses-json --group futhark-server --group gradbench
 
+# Install Futhark.
 ADD https://futhark-lang.org/releases/futhark-0.25.28-linux-x86_64.tar.xz futhark-0.25.28-linux-x86_64.tar.xz
-
 RUN tar xvf futhark-0.25.28-linux-x86_64.tar.xz
-
 RUN make -C futhark-0.25.28-linux-x86_64 install
 
-COPY python /gradbench/python
-ENV PYTHONPATH=/gradbench/python/gradbench
+# Install Futhark packages.
+COPY tools/futhark tools/futhark
+RUN cd tools/futhark && futhark pkg sync
 
-COPY tools/futhark /gradbench/tools/futhark
-RUN cd /gradbench/tools/futhark && futhark pkg sync
+# `COPY` the rest of the files.
+COPY python python
 
-WORKDIR /gradbench
-ENTRYPOINT ["python", "/gradbench/tools/futhark/run.py"]
+ENTRYPOINT ["uv", "run", "--no-sync", "python/gradbench/gradbench/cpp.py", "futhark"]
```

Comment:
This I am not pro. Or at least I don't understand how this doesn't break the command line argument parsing implemented in `tools/futhark/run.py` (and for that matter how it can work at all, since the Futhark tool isn't even close to working the way the C++ tools do).

Reply:
@athas sorry, it looks like you got to this line before I fixed it. Can you look again?